tests/command_list_test.rs

```diff
@@ -1,15 +1,52 @@
+use std::collections::HashMap;
+
 use assert_cmd::Command;
+use juliaup::utils::get_juliaserver_base_url;
+use juliaup::versions_file::*;
 use predicates::prelude::*;
 
 #[test]
 fn command_list() {
-    let depot_dir = tempfile::Builder::new().prefix("juliauptest").tempdir().unwrap();
+    let depot_dir = tempfile::Builder::new()
+        .prefix("juliauptest")
+        .tempdir()
+        .unwrap();
 
     Command::cargo_bin("juliaup")
-        .unwrap()        
+        .unwrap()
         .arg("list")
         .env("JULIA_DEPOT_PATH", depot_dir.path())
         .assert()
         .success()
         .stdout(predicate::str::starts_with(" Channel").and(predicate::str::contains("release")));
 }
+
+#[test]
+fn version_url() {
+    let up_server_json_url = get_juliaserver_base_url()
+        .unwrap()
+        .join("./bin/versions.json");
+    let json_versions: JsonVersion = ureq::get(up_server_json_url.unwrap().as_str())
+        .call()
+        .unwrap()
+        .into_json()
+        .unwrap();
+    let mut exist_urls: HashMap<String, bool> = HashMap::new();
+    for (_, v) in json_versions {
+        for f in v.files {
+            exist_urls.insert(f.url, true);
+        }
+    }
+
+    let vdb = load_versions_db().unwrap();
+    for (_, v) in vdb.available_versions {
+        let full_url = format!(
+            "https://julialang-s3.julialang.org/{}",
+            v.url_path.to_string()
+        );
+        if !exist_urls.contains_key(&full_url) {
+            println!("{}", &full_url);
```

Comment:
Could you add a comment or two regarding what you are testing here?

Also, unless you instruct someone to run the tests with `--nocapture`, this `println!`s will not show up.
Is this intended?

Reply:
I initially thought every url should be in the versions.json. But as it is not the case, I just added the code to show the diff and wait for your opinions.

Reply:
Hmm. Interesting point. I don't the reason behind this. Based on a quick screening, it does not look like any actual post-1.0-versions are missing. All of the entries seem to be RCs and alpha versions.
I'd guess that it's unlikely that people will require these versions, if the actually released versions seem to be available.

About the pre-1.0 versions I'm not sure. @davidanthoff Do you have any ideas on this? Are the missing Windows versions going to be an issue?